cli/azd/cmd/cobra_builder.go

```diff
@@ -215,6 +215,32 @@ func (cb *CobraBuilder) bindCommand(cmd *cobra.Command, descriptor *actions.Acti
 		}
 	}
 
+	if descriptor.Options.GroupingOptions.RootLevelHelp != actions.CmdGroupNone {
+		if cmd.Annotations == nil {
+			cmd.Annotations = make(map[string]string)
+		}
+		switch descriptor.Options.GroupingOptions.RootLevelHelp {
+		case actions.CmdGroupConfig:
+			cmd.Annotations[string(cmdGrouperKey)] = string(cmdGroupConfig)
+		case actions.CmdGroupManage:
+			cmd.Annotations[string(cmdGrouperKey)] = string(cmdGroupManage)
+		case actions.CmdGroupAbout:
+			cmd.Annotations[string(cmdGrouperKey)] = string(cmdGroupAbout)
+		case actions.CmdGroupMonitor:
+			cmd.Annotations[string(cmdGrouperKey)] = string(cmdGroupMonitor)
+		}
+	}
+
```

Comment:
It feels like our custom grouping of commands shouldn't need to leak down into the generic cobra command builder.

Reply:
Can we use the natural nesting structure of the action descriptors to dynamically figure out the groupings?

Reply:
do you mean, creating a hardcoded relation where we can check where each command belongs to?
Or what do you mean?

Reply:
I just don't know why we need this switch statement with leaked information about our cmd groups in this generic code.  What are the alternatives?

Reply:
This is the piece I don't like.  Why do we need the switch statement at all and leek our groups here? Can't we just assign the annotation grouper key to the string value of `descriptor.Options.GroupingOptions.RootLevelHelp` ??

```suggestion
	if descriptor.Options.GroupingOptions.RootLevelHelp != actions.CmdGroupNone {
		if cmd.Annotations == nil {
			cmd.Annotations = make(map[string]string)
		}
		cmd.Annotations[string(cmdGrouperKey)] = string(descriptor.Options.GroupingOptions.RootLevelHelp)
	}

```

Reply:
This is a side effect of having the package `actions` as a subpackage of `cmd`
The action descriptor belongs to the `actions` pgk.  So I had to define the grouping types within the `actions` pkg.

I created an issue in `azure-dev-pr` where I am proposing to get rid of the `actions` pkg by merging it to the `cmd` pgk.

Then I wouldn't need to have this `switch` to convert from types from `actions` to types from `cmd` pkg.

Reply:
I think, as an alternative, I can move the gropup names and types to be only in the `actions` pkg...

Reply:
updated